src/app/auth.guard.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+      if(localStorage.getItem('isLoggedin') === 'true') {
```

Comment:
@BilalAlam173 please maintain the user state in user.service and fetch it from local storage once in the beginning